core/primitives/src/runtime/parameter_table.rs

```diff
@@ -139,22 +218,27 @@ fn parse_parameter_txt_value(value: &str) -> Result<serde_json::Value, InvalidCo
         return Ok(serde_json::Value::Null);
     }
     if value.chars().all(|c| c.is_numeric() || c == '_') {
-        Ok(serde_json::Value::Number(
-            value
-                .chars()
-                .filter(|c| c.is_numeric())
-                .collect::<String>()
-                .parse()
-                .map_err(|err| InvalidConfigError::ValueParseError(err, value.to_owned()))?,
-        ))
+        let raw_number = value.chars().filter(|c| c.is_numeric()).collect::<String>();
+        // We do not have "arbitrary_precision" serde feature enabled, thus we
+        // can only store up to `u64::MAX`, which is `18446744073709551615` and
+        // has 20 characters.
+        if raw_number.len() < 20 {
```

Comment:
Are we happy with the behaviour here when somebody specifies 10^19 which is 20 digits long but is still 2<sup>63.11</sup> and fits into u64? `serde_json` does have an error code for out-of-range numbers internally but sadly it isn't exposed :(

If not, I'm worried that serde_json might not actually be the right tool for us tbh.

Reply:
It probably isn't terribly difficult to implement a custom deserializer just for the values. An example of an implementation that does this is the aforementioned [`config-rs`](https://github.com/mehcode/config-rs/blob/master/src/de.rs) (I encourage to at least skim throug the link here). It has handling for u128 built-in too.

Reply:
Ok I see, a custom value parser would make sense. Also thanks for the link with the example.

As things stand right now, I don't see a strong motivation to do it right now. As long as we have JSON as intermediary, we have to store numbers as strings anyway and as long as we don't lose precision on the way, I think it doesn't matter too much where the cut-off point is.